[3/5] Put JIT source inside flatbuffer

qihqi opened the pull request:
Summary: title

Test Plan: unittest

Differential Revision: D34881612

iseeyuan commented on test/cpp/jit/test_flatbuffer.cpp:
```diff
@@ -1135,5 +1135,91 @@ TEST(FlatbufferTest, OperatorTest2) { // NOLINT (use =delete in gtest)
+  Module m2 = jitModuleFromBuffer(data.data());
+  auto res = m2.forward(inputs);
+
+  auto resd = res.toTensor();
```
Would be nice to add a mobile module from the same data and call forward().

qihqi replied:
done

iseeyuan commented on test/cpp/jit/test_flatbuffer.cpp:
```diff
@@ -1135,5 +1135,91 @@ TEST(FlatbufferTest, OperatorTest2) { // NOLINT (use =delete in gtest)
   }
 }
 
+Module jitModuleFromBuffer(void* data) {
```
Would it be parse_and_initialize_jit_module() below?

qihqi replied:
yeah though if doing so would need to transfer ownership from flatbuffers::DetachedBuffer to mobile::Module so it's bit annoying.

iseeyuan reviewed (approved):
LGTM

qihqi committed:
[3/5] Put JIT source inside flatbuffer (#74245)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/74245

title

Test Plan: unittest

Reviewed By: iseeyuan

Differential Revision: D34881612

fbshipit-source-id: 32ece5afbe68141533d9d091e99527556cb281ca
